packages/editor-ui/src/plugins/telemetry/index.ts

```diff
@@ -0,0 +1,52 @@
+import _Vue from "vue";
+import {
+	ITelemetrySettings,
+	IDataObject,
+} from 'n8n-workflow';
+
+import * as rudderanalytics from 'rudder-sdk-js';
+
+declare module 'vue/types/vue' {
+	interface Vue {
+		$telemetry: Telemetry;
+	}
+}
+
+export function TelemetryPlugin(vue: typeof _Vue): void {
+	const telemetry = new Telemetry();
+
+	Object.defineProperty(vue, '$telemetry', {
+		get() { return telemetry; },
+	});
+	Object.defineProperty(vue.prototype, '$telemetry', {
+		get() { return telemetry; },
+	});
+}
+
+class Telemetry {
+
+	private telemetry?: any; // tslint:disable-line:no-any
+
+	init(options: ITelemetrySettings) {
+		if (options.enabled && !this.telemetry) {
+			if(!options.config) {
+				return;
+			}
+
+			rudderanalytics.load(options.config.key, options.config.url, { logLevel: 'DEBUG' });
+			this.telemetry = rudderanalytics;
+		}
+	}
+
+	identify(event: string, properties?: IDataObject) {
+		if (this.telemetry) {
+			this.telemetry.identify(event, properties);
+		}
+	}
+
+	track(event: string, properties?: IDataObject) {
+		if (this.telemetry) {
+			this.telemetry.track(event, properties);
+		}
+	}
```

Comment:
Maybe error out if `this.telemetry` is uninitialized?

Reply:
we can, but we want the tracking code to run seamlessly even when telemetry is disabled